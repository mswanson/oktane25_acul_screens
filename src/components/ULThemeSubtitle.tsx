import type { HTMLAttributes, ReactNode } from "react";

import { cn } from "@/lib/utils";

export interface ULThemeSubtitleProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * The subtitle of the screen.
   */
  children: ReactNode;
  /**
   * Optional class names for additional styling or overriding default styles.
   */
  className?: string;
}

const ULThemeSubtitle = ({
  children,
  className,
  ...rest
}: ULThemeSubtitleProps) => {
  const themedStyles =
    "mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle";

  return (
    <p className={cn(themedStyles, className)} {...rest}>
      {children}
    </p>
  );
};

export default ULThemeSubtitle;
