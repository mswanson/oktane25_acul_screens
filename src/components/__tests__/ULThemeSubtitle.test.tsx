import { render } from "@testing-library/react";

import ULThemeSubtitle from "../ULThemeSubtitle";

describe("ULThemeSubtitle Component", () => {
  // Snapshot Tests
  it("matches snapshot with default props", () => {
    const { container } = render(
      <ULThemeSubtitle>Test Subtitle</ULThemeSubtitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with custom className", () => {
    const { container } = render(
      <ULThemeSubtitle className="custom-class">Test Subtitle</ULThemeSubtitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with additional props", () => {
    const { container } = render(
      <ULThemeSubtitle id="test-id" data-testid="subtitle">
        Test Subtitle
      </ULThemeSubtitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with multiple children", () => {
    const { container } = render(
      <ULThemeSubtitle>
        <span>Child 1</span>
        <span>Child 2</span>
      </ULThemeSubtitle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Fuctional Tests

  it("renders children correctly", () => {
    const { getByText } = render(
      <ULThemeSubtitle>Test Subtitle</ULThemeSubtitle>
    );
    expect(getByText("Test Subtitle")).toBeInTheDocument();
  });

  it("applies additional props correctly", () => {
    const { container } = render(
      <ULThemeSubtitle data-testid="subtitle">Test Subtitle</ULThemeSubtitle>
    );
    expect(container.firstChild).toHaveAttribute("data-testid", "subtitle");
  });

  it("combines themed styles and custom className", () => {
    const { container } = render(
      <ULThemeSubtitle className="custom-class">Test Subtitle</ULThemeSubtitle>
    );
    expect(container.firstChild).toHaveClass(
      "mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle"
    );
    expect(container.firstChild).toHaveClass("custom-class");
  });

  it("combines themed styles and custom className", () => {
    const { container } = render(
      <ULThemeSubtitle className="custom-class">Test Subtitle</ULThemeSubtitle>
    );
    expect(container.firstChild).toHaveClass(
      "mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle"
    );
    expect(container.firstChild).toHaveClass("custom-class");
  });
});
