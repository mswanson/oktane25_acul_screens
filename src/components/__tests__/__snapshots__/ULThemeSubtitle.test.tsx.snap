// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemeSubtitle Component matches snapshot with additional props 1`] = `
<p
  class="mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle"
  data-testid="subtitle"
  id="test-id"
>
  Test Subtitle
</p>
`;

exports[`ULThemeSubtitle Component matches snapshot with custom className 1`] = `
<p
  class="mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle custom-class"
>
  Test Subtitle
</p>
`;

exports[`ULThemeSubtitle Component matches snapshot with default props 1`] = `
<p
  class="mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle"
>
  Test Subtitle
</p>
`;

exports[`ULThemeSubtitle Component matches snapshot with multiple children 1`] = `
<p
  class="mb-4 text-body-text justify-text-header text-(length:--ul-theme-font-subtitle-size) font-subtitle"
>
  <span>
    Child 1
  </span>
  <span>
    Child 2
  </span>
</p>
`;
